packages/database/src/migrations/20211006223410-ConvertFanfanaOlaHpuOverlaysToReportServer-modifies-data.js

```diff
@@ -0,0 +1,258 @@
+'use strict';
+
+import { insertObject, nameToId, codeToId, updateValues, generateId } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+
+const PERMISSION_GROUP = 'Tonga Health Promotion Unit';
+const HPU_GROUP_CODE = 'Fanafana_Ola_HPU';
+const NEW_GROUP_VALUES = {
+  name: 'Health Promotion Unit',
+  code: 'Health_Promotion_Unit',
+};
+const LEGACY_OVERLAY_CODES = [
+  'TO_HPU_NCD_Risk_Factory_Screening_Type_Setting_Unique',
+  'TO_HPU_Health_Talks_Training_Type_Setting_Unique',
+  'TO_HPU_Physical_Activity_Type_Setting_Unique',
+];
+
+const NEW_OVERLAY_CONFIGS = [
+  {
+    reportCode: 'FANAFANA_hpu_ncd_risk_factor_screening_setting_type_map',
+    dataElements: ['HP31n'],
+    name: 'NCD Risk Factor Screening: Setting Type',
+    dataGroups: ['HP02'],
+    sortOrder: 0,
+  },
+  {
+    reportCode: 'FANAFANA_hpu_health_talks_training_type_setting_map',
+    dataElements: ['HP281'],
+    name: 'Health Talks and Training: Setting Type',
+    dataGroups: ['HP07'],
+    sortOrder: 1,
+  },
+  {
+    reportCode: 'FANAFANA_hpu_physical_activity_type_setting_map',
+    dataElements: ['HP4'],
+    name: 'Physical Activity: Setting Type',
+    dataGroups: ['HP01'],
+    sortOrder: 2,
+  },
+];
+
+const getReportObject = (reportCode, dataElements, dataGroups) => ({
+  code: reportCode,
+  config: {
+    fetch: {
+      dataElements,
+      dataGroups,
+    },
+    transform: [
+      {
+        transform: 'updateColumns',
+        exclude: ['event', 'eventDate', 'orgUnitName', 'orgUnit', dataElements[0]],
+        insert: {
+          organisationUnitCode: '=$orgUnit',
+          Church: `= eq($${dataElements[0]},'Church') ? 1 : undefined`,
+          Community: `= eq($${dataElements[0]},'Community') ? 1 : undefined`,
+          Workplace: `= eq($${dataElements[0]},'Workplace') ? 1 : undefined`,
+          School: `= eq($${dataElements[0]},'School') ? 1 : undefined`,
+          Unique: `=$${dataElements[0]}`,
```

Comment:
```suggestion
        insert: {
          organisationUnitCode: '=$orgUnit',
          `=$${dataElements[0]}`: `= 1`,
          Unique: `=$${dataElements[0]}`,
```
I believe this works?

Reply:
Looks like it only works within square brackets for the key [`=$${dataElements[0]}`] @rohan-bes